src/views/image/RegisterIso.vue

```diff
@@ -0,0 +1,231 @@
+            <a-select-option v-for="(opt, optIndex) in osTypes" :key="optIndex">
+              {{ opt.name || opt.description }}
+            </a-select-option>
+          </a-select>
+        </a-form-item>
+
+        <a-form-item :label="$t('isextractable')">
+          <a-switch v-decorator="['isextractable']" />
+        </a-form-item>
+
+        <a-form-item :label="$t('ispublic')">
+          <a-switch v-decorator="['ispublic']" />
+        </a-form-item>
+
+        <a-form-item :label="$t('isfeatured')">
+          <a-switch v-decorator="['isfeatured']" />
+        </a-form-item>
+
+        <div :span="24" class="action-button">
+          <a-button @click="closeAction">{{ this.$t('Cancel') }}</a-button>
+          <a-button :loading="loading" type="primary" @click="handleSubmit">{{ this.$t('OK') }}</a-button>
+        </div>
+      </a-form>
+    </a-spin>
+  </div>
+</template>
+
+<script>
+import { api } from '@/api'
+
+export default {
+  name: 'RegisterIso',
+  data () {
+    return {
+      zones: [],
+      osTypes: [],
+      zoneLoading: false,
+      osTypeLoading: false,
+      loading: false
+    }
+  },
+  beforeCreate () {
+    this.form = this.$form.createForm(this)
+  },
+  created () {
+    this.zones = [
+      {
+        id: '-1',
+        name: this.$t('label.all.zone')
+      }
+    ]
+  },
+  mounted () {
+    this.fetchData()
+  },
+  methods: {
+    fetchData () {
+      this.fetchZoneData()
+      this.fetchOsType()
+    },
+    fetchZoneData () {
+      const params = {}
+      params.listAll = true
+
+      this.zoneLoading = true
+
+      api('listZones', params).then(json => {
+        const listZones = json.listzonesresponse.zone
+        this.zones = this.zones.concat(listZones)
+      }).finally(() => {
+        this.zoneLoading = false
+      })
+    },
+    fetchOsType () {
+      const params = {}
+      params.listAll = true
+
+      this.osTypeLoading = true
+
+      api('listOsTypes', params).then(json => {
+        const listOsTypes = json.listostypesresponse.ostype
+        this.osTypes = this.osTypes.concat(listOsTypes)
+      }).finally(() => {
+        this.osTypeLoading = false
+      })
+    },
+    handleSubmit (e) {
+      e.preventDefault()
+      this.form.validateFields((err, values) => {
+        if (err) {
+          return
+        }
+
+        const params = {}
+        for (const key in values) {
+          const input = values[key]
+
+          if (input === undefined) {
+            continue
+          }
+
+          switch (key) {
+            case 'zoneid':
+              params[key] = this.zones[input].id
+              break
+            case 'ostypeid':
+              params[key] = this.osTypes[input].id
+              break
+            default:
+              params[key] = input
+              break
+          }
+        }
+
+        this.loading = true
+
+        api('registerIso', params).then(json => {
+          console.log(json)
```

Comment:
@utchoang here we can call `this.$pollJob` to check/poll for the registerIso task.

Reply:
Ignore just checked, the API is not async API.